### PR TITLE
fix scheduler scheduling & preempting attempt metrics

### PR DESCRIPTION
**Pull request by lzlaa**
In the current implementation, when the Scheduling and Preempting stages fail, both `scheduler_pod_scheduling_attempts{unschedulable}` and `scheduler_pod_preempting_attempts{nominatedFailure}` will be recorded, leading to unclear semantics for these metrics.

In this MR:
- when Scheduling and Preempting both fail: `scheduler_pod_scheduling_attempts{unschedulable} += 1`; `scheduler_pod_preempting_attempts{nominatedFailure} += 1`
- when Scheduling fails but Preempting succeeds: `scheduler_pod_scheduling_attempts{scheduled} += 1`; `scheduler_pod_preempting_attempts{nominated} += 1`

### CODE REVIEW

File: pkg/scheduler/framework/unit_runtime/unit_framework.go [hunk 1]
```diff
@@ -348,8 +348,9 @@ func (f *UnitFramework) scheduleOneUnitInstance(ctx context.Context, scheduledIn
 	runningUnitInfo *core.RunningUnitInfo, unitKey string, queuePriorityScore float64, unitCycleState, commonPreemptionState *framework.CycleState,
 	nodeGroup framework.NodeGroup,
 	usr *framework.UnitSchedulingRequest,
-) (bool, error) {
+) (success bool, err error) {
 	switchType, subCluster := f.handle.SwitchType(), f.handle.SubCluster()
+	godelScheduler := f.schedulerHooks.PodScheduler()
```
Comment (binacs) [inline pkg/scheduler/framework/unit_runtime/unit_framework.go:353]: Why is this initialization being advanced? If it is necessary to do, should we modify `preemptOneUnitInstance` as well?

**Review by binacs — APPROVED** [review]

**Comment by zryfish** [comment]
@lzlaa Please rebase and push again

**Comment by NickrenREN** [comment]
any new comments on this PR ? @zryfish @binacs

**Review by binacs — APPROVED** [review]

**Comment by binacs** [comment]
lgtm

**Pull request merged**